nanopc-t6.mk

```diff
@@ -0,0 +1,261 @@
+
+COMPILE_NS_USER ?= 64
+override COMPILE_NS_KERNEL := 64
+COMPILE_S_USER ?= 64
+COMPILE_S_KERNEL ?= 64
+
+include common.mk
+
+DEBUG ?= 1
+
+# Do not leave a partially downloaded binary in case wget fails midway
+.DELETE_ON_ERROR:
+
+################################################################################
+# Paths to git projects and various binaries
+################################################################################
+TF_A_PATH		?= $(ROOT)/trusted-firmware-a
+BINARIES_PATH		?= $(ROOT)/out
+UBOOT_PATH		?= $(ROOT)/u-boot
+UBOOT_BIN		?= $(UBOOT_PATH)/u-boot.bin
+ROOT_IMG		?= $(ROOT)/out-br/images/rootfs.ext2
+BOOT_IMG		?= $(ROOT)/out/nanopc-t6.img
+TPL_BIN			?= $(BINARIES_PATH)/rk3588_ddr_lp4_2112MHz_lp5_2400MHz_v1.18.bin
+BOOT_CMD_PATCH		?= $(ROOT)/build/nanopc-t6/nanopi6.h.patch
+BOARD_DTSI_PATCH	?= $(ROOT)/build/nanopc-t6/rk3588-nanopi6-common.dtsi.patch
+LINUX_DTSI		?= $(LINUX_PATH)/arch/arm64/boot/dts/rockchip/rk3588-nanopi6-common.dtsi
+UBOOT_HEADER		?= $(UBOOT_PATH)/include/configs/nanopi6.h
+
+LINUX_MODULES ?= n
+
+BR2_TARGET_ROOTFS_CPIO = n
+BR2_TARGET_ROOTFS_CPIO_GZIP = n
+BR2_TARGET_ROOTFS_EXT2 = y
+# Use Debug UART (tty1) for system console
+# Reference: https://wiki.friendlyelec.com/wiki/index.php/NanoPC-T6#NanoPC-T6
+BR2_TARGET_GENERIC_GETTY_PORT = tty1
+ifeq ($(LINUX_MODULES),y)
+# If modules are installed...
+# ...enable automatic device detection and driver loading
+BR2_ROOTFS_DEVICE_CREATION_DYNAMIC_EUDEV = y
+# ...and configure eth0 automatically based on ifup helpers
+BR2_PACKAGE_IFUPDOWN_SCRIPTS = y
+#BR2_SYSTEM_DHCP = eth0
+# An image with module takes more space
+BR2_TARGET_ROOTFS_EXT2_SIZE = 1536M
+# Enable SSH daemon for remote login
+BR2_PACKAGE_OPENSSH = y
+BR2_PACKAGE_OPENSSH_SERVER = y
+BR2_ROOTFS_POST_BUILD_SCRIPT = $(ROOT)/build/br-ext/board/nanopc-t6/post-build.sh
+# Enable NTP for current time
+BR2_PACKAGE_NTP = y
+BR2_PACKAGE_NTP_NTPD = y
+else
+BR2_TARGET_ROOTFS_EXT2_SIZE = 112M
+endif
+
+################################################################################
+# Targets
+################################################################################
+
+all: boot-img
+
+clean: buildroot-clean
+
+include toolchain.mk
+
+################################################################################
+# Arm Trusted Firmware-A
+################################################################################
+TF_A_EXPORTS ?= CROSS_COMPILE="$(CCACHE)$(AARCH64_CROSS_COMPILE)" \
+		M0_CROSS_COMPILE="$(CCACHE)$(AARCH32_CROSS_COMPILE)"
```

Comment:
The latest version of TF-A requires `CC` and `LD` variables like:
https://github.com/OP-TEE/build/blob/009f86a824c6b1653a22d422696c452c77ab9f85/qemu_v8.mk#L149-L150

I don't know how that affects the `M0_CROSS_COMPILE`.